### ISSUE TITLE
Carousel显示不正确

### ISSUE THREAD

**Issue by WCKYWCKF** [irihitech/Semi.Avalonia#512]
一个BUG，但可能和Semi无关。
使用Shift+鼠标滚轮切换Carousel的轮播图时装饰器没有对应Content的Index。
![00260d5109e61f6b80a0f89c011c9a73](https://github.com/user-attachments/assets/329a8c2f-a4b5-49cf-ad6d-8d726c08ab5d)

**Comment by rabbitism** [comment]
Yes, Carousel的ScrollViewer offset变化的时候不会反过来更新Index，我来想想这个有什么办法在上游修复。

**Comment by rabbitism** [comment]
非常神奇，只要在Template中存在其他元素就会导致ScrollViewer相应鼠标滚动动作……

**Comment by WCKYWCKF** [comment]
> 非常神奇，只要在Template中存在其他元素就会导致ScrollViewer相应鼠标滚动动作……

Carousel的本质是一个ItemsPresenter，它通过ScrollViewer.Offset控制显示哪一个Item。
![image](https://github.com/user-attachments/assets/08da5218-9461-46a4-9247-46eed7d556b4)
我认为OnApplyTemplate（并在脱离时取消订阅）中添加对ScrollViewer的PointerWheelChanged订阅来反向更改Carousel.SelectedIndex是可以缓解此问题的。

吐槽：在TextBox中使用shift+鼠标滚轮让整个UI页面一片空白令我沉默（因为某个不可知的BUG要重构整个UI了，我用TransitioningContentControl替换了Carousel）

**Comment by rabbitism** [comment]
> 我认为OnApplyTemplate（并在脱离时取消订阅）中添加对ScrollViewer的PointerWheelChanged订阅来反向更改 
> Carousel.SelectedIndex是可以缓解此问题的。
> 
> 吐槽：在TextBox中使用shift+鼠标滚轮让整个UI页面一片空白令我沉默（因为某个不可知的BUG要重构整个UI了，我用TransitioningContentControl替换了Carousel）

离谱的点在于说，如果是普通的Carousel默认主题，那么是根本不会响应Wheel事件的。一点ScrollViewer上多遮罩了其他元素，那就会触发ScrollPresenter的Wheel事件

**Comment by WCKYWCKF** [comment]
> 离谱的点在于说，如果是普通的Carousel默认主题，那么是根本不会响应Wheel事件的。一点ScrollViewer上多遮罩了其他元素，那就会触发ScrollPresenter的Wheel事件

我理解了。这确实很离谱，令人沉默

**Comment by WCKYWCKF** [comment]
@rabbitism 修好了。并且，我认为这与Avalonia的特性Background="Transparent"有关。因为只有我的鼠标指针处于`1`上时滑动滚轮才会触发Wheel事件。
![image](https://github.com/user-attachments/assets/27908d28-a44d-4c2c-9860-d0f80a5f27af)
